### PR TITLE
Meteor wave changes.

### PR DESCRIPTION
**Pull request by PowerfulBacon**
## About The Pull Request

Meteors will now be warned of 10 minutes before they strike.
When meteors are detected on a collision course with the station, meteor shields will be unlocked.

## Why It's Good For The Game

Adds in an intense window to save yourself from the meteors, gives engineers and cargo an opportunity to cooperate and lessens the impact of meteors on longer rounds.

## Changelog
:cl:
tweak: Meteors now spawn 10 minutes after being announced.
tweak: Meteor wave event now unlocks meteor shields.
tweak: Meteor wave start time reduced from 50 minutes to 30 minutes.
/:cl:

### CODE REVIEW

File: code/modules/events/meteor_wave.dm [hunk 1]
```diff
@@ -47,7 +47,11 @@
 			kill()
 
 /datum/round_event/meteor_wave/announce(fake)
-	priority_announce("Meteors have been detected on collision course with the station.", "Meteor Alert", 'sound/ai/meteors.ogg')
+	priority_announce("Meteors have been detected on collision course with the station. Estimated time until impact: 10 MINUTES. Anti-meteor point defense is available for purchase via the station's cargo shuttle.", "Meteor Alert", 'sound/ai/meteors.ogg')
+	var/datum/supply_pack/P = SSshuttle.supply_packs[/datum/supply_pack/engineering/shield_sat]
+	P.special_enabled = TRUE
+	P = SSshuttle.supply_packs[/datum/supply_pack/engineering/shield_sat_control]
+	P.special_enabled = TRUE
```
Comment (ike709) [inline code/modules/events/meteor_wave.dm:54]: hmmmm

**Comment by NickEmbased** [comment]
Nice :+1:

**Comment by That0nePerson** [comment]
Amazing idea

**Comment by OceanFish1** [comment]
No round ending meteors? Thats something

**Comment by ike709** [comment]
I like it

**Comment by utilbird** [comment]
There should be an admin option to bypass or alter the timer

**Comment by PowerfulBacon** [comment]
> 
> 
> There should be an admin option to bypass or alter the timer

varedit

**Comment by eeSPee** [comment]
Can you also somehow call out the direction they are going to come from?

**Comment by PowerfulBacon** [comment]
there is no direction, its from all over. This is more of a way to mitigate damage to important areas than to completely stop the meteors, obviously, 10 minutes isn't long enough to cover the entire station unless everyone in the crew works together and all the budgets are aquired.

**Comment by eeSPee** [comment]
> there is no direction, its from all over. 

No, they spawn from a direction. They don't ask for an argument direction, but meteors usually spawn from a direction.

> This is more of a way to mitigate damage to important areas than to completely stop the meteors, obviously, 10 minutes isn't long enough to cover the entire station unless everyone in the crew works together and all the budgets are aquired.

It's more than enough. Honestly may make meteors way more interesting.

**Comment by utilbird** [comment]
> varedit

Something more user friendly, like how rod warnings/special targets work

**Comment by park66665** [comment]
Just code a simple input code via `/datum/round_event_control/meteor_wave/admin_setup` if admins want greater adminbus, like altering the timer. However, I am not entirely convinced of making it admin-friendly.

**Comment by Pontenerd** [comment]
> 
> 
>  I am not entirely convinced of making it admin-friendly.

Why's that then?

**Comment by park66665** [comment]
> Why's that then?

Excessive adminbus through meteor waves happening without delay. That's why.

**Comment by ike709** [comment]
> Excessive adminbus through meteor waves happening without delay. That's why.

But this is how I, ahem, _encourage_ roundends whenever I need the server to update.

**Comment by PowerfulBacon** [comment]
oh yea I forgot about throwing meteors at the station with the intent to destroy it.

**Comment by DatBoiTim** [comment]
> Just code a simple input code via `/datum/round_event_control/meteor_wave/admin_setup` if admins want greater adminbus, like altering the timer. However, I am not entirely convinced of making it admin-friendly.

Well making it take a custom length of time can actually be good for admins who, say, want to run a colonization event and don't want to have to keep an eye on the timer, and instead just set a 30 minute timer meteor shower of whatever severity.

**Comment by PowerfulBacon** [comment]
Fine how about a giant super meteor event that fast calls the shuttles and then nukes the station

**Comment by utilbird** [comment]
> asteroid impact doesn't send a 256x256 asteroid through the station

😢

**Review by utilbird — APPROVED** [review]

**Comment by wraith-54321** [comment]
you know, thing is, whats gonna end up happening is that now the shuttle is gonna get called every time

**Comment by ike709** [comment]
Admins no longer have the ability to trigger meteors instantly

**Pull request merged**